lockfile/src/lib.rs

```diff
@@ -185,6 +193,7 @@ mod tests {
             ("gradle", LockfileFormat::Gradle),
             ("nuget", LockfileFormat::Msbuild),
             ("msbuild", LockfileFormat::Msbuild),
+            ("gosum", LockfileFormat::GoSum),
```

Comment:
Do Go developers actually refer to this as "gosum" or is the "sum" here basically just like the ".lock" suffix in other ecosystems?

If it's just the standard lockfile format for go, we might want to consider just naming it "go" so users can more intuitively pick their desired format.

Reply:
I went back and forth on this myself, I'm open to change. I picked `GoSum` because:

- Go has had a number of different dependency management systems over the years. I think it is likely this latest one will remain the defacto standard, but there has been enough change that it seems prudent to be specific with the name (and who knows, maybe we will end up supporting one of the older systems also)
- Similar to other languages/ecosystems, there's a package file `go.mod` and lock file `go.sum`. Some of the other languages here parse things that are more like a package file than a lock file, and I when we were discussing go there was some debate about also parsing `go.mod` files, so I think the name should leave that option open.

Reply:
(and fwiw I think that go developers do say "go sum file", but I'm not deeply into the community so I don't really know)

Reply:
Fair enough, having had other systems might be reason enough to be more explicit.

Though the difference between `.mod` and `.sum` isn't really one that matters considering the same could be said about `packge-lock.json`, about yarn's config, about gradle, ... . Pretty much every package manager that has a lockfile also has an "unlocked" manifest. We're more pointing out the package management system here than pointing out the lockfile specifically.

Reply:
> Though the difference between .mod and .sum isn't really one that matters considering the same could be said about packge-lock.json, about yarn's config, about gradle,

I agree with this. And the name should probably just be `go` for now.

This may get complicated in the future if we allow submission of either `go.mod` or `go.sum` files (See phylum-dev/roadmap#173). But we're going to have to deal with that across the board, so it's not a reason to deviate from normal now.

Reply:
Are lockfiles not standard for Go the way they are for Rust for example? Generally I'd avoid accepting non-lock files wherever possible. Nobody should **ever** submit a `Cargo.toml` for analysis.

Reply:
Fair enough, I changed to `go`!